### PR TITLE
Fix dynamic assembly memory leak, optimize Regex usage

### LINKED ISSUE #142: Dynamic Assembly memory leak
There is a memory leak in the implementation of `XmlSerializer` usage.  A new `XmlSerializer(typeof(SaveData))` is created every time the following are called:
- `BarcodeLib.ToXML()`
- `BarcodeLib.FromXML()`

This usage creates a dynamic assembly every time the `XmlSerailizer(typeof(T))` constructor is invoked. This results in a memory leak.

**EDIT:** _This usage creates a dynamic assembly every time the `XmlSerailizer(typeof(T))` constructor is invoked._ This is not correct. The leak occurs in different constructors.  I've been so used to having to correct these in other apps, I confused the dynamic assembly memory leak with a different constructor. My apologies.

### PR DESCRIPTION
**Pull request by udlose**
Fixes #142, #144
- fix dynamic assembly memory leak by changing `XmlSerializer` to be static
- fix code analysis warning CA1816
- performance optimization for Regex usage (reduces cpu time, memory allocations)

### CODE REVIEW

File: BarcodeStandard/BarcodeLib.cs [hunk 1]
```diff
@@ -1101,10 +1102,9 @@ public string ToXML(Boolean includeImage = true)
                 {
                     using (SaveData xml = GetSaveData(includeImage))
                     {
-                        XmlSerializer writer = new XmlSerializer(typeof(SaveData));
```
Comment (barnhill) [inline BarcodeStandard/BarcodeLib.cs:1104]: Should we just wrap this with a using as the static never gets GCed?

File: BarcodeStandard/BarcodeLib.cs [hunk 2]
```diff
@@ -1332,8 +1331,8 @@ void IDisposable.Dispose()
         {
             // Do not change this code. Put cleanup code in Dispose(bool disposing) above.
             Dispose(true);
-            // TODO: uncomment the following line if the finalizer is overridden above.
-            // GC.SuppressFinalize(this);
+            // Call GC.SuppressFinalize(object). This will prevent derived types that introduce a finalizer from needing to re-implement 'IDisposable' to call it.	
+            GC.SuppressFinalize(this);
```
Comment (barnhill) [inline BarcodeStandard/BarcodeLib.cs:1335]: Will this promote bad behavior by not making implementations think through disposal?

Comment (udlose) [reply BarcodeStandard/BarcodeLib.cs:1335]: I would hope not.  IMO, developers should never rely on the implementation details of an underlying library.  If it implements `IDisposable`, it should be disposed.  Unless a developer looked inside the code, they wouldn't even know.

Comment (udlose) [reply BarcodeStandard/BarcodeLib.cs:1335]: BTW, I wrote something about not calling `Dispose` some years ago on SO - [Dispose, when is it called?](https://stackoverflow.com/questions/2871888/dispose-when-is-it-called/9432317#9432317) and [What happens if I don't call Dispose on the pen object?](https://stackoverflow.com/questions/4267729/what-happens-if-i-dont-call-dispose-on-the-pen-object/5555243#5555243)

**Review by barnhill — APPROVED** [review]

**Pull request merged**